src/cmd/run/canary_mode.rs

```diff
@@ -0,0 +1,68 @@
+use async_trait::async_trait;
+use miette::Result;
+use tokio::time::Duration;
+use tokio_graceful_shutdown::{IntoSubsystem as _, SubsystemBuilder, Toplevel};
+use tracing::{debug, info};
+
+use crate::ControllerSubsystem;
+use crate::adapters::{BackendClient, BoxedIngress, BoxedMonitor, BoxedPlatform, RolloutMetadata};
+use crate::subsystems::CONTROLLER_SUBSYSTEM_NAME;
+
+use super::{DEFAULT_SHUTDOWN_TIMEOUT, DeploymentMode};
+
+/// Canary deployment mode - runs the full canary analysis subsystems
+pub struct CanaryMode {
+    backend: BackendClient,
+    monitor: BoxedMonitor,
+    ingress: BoxedIngress,
+    platform: BoxedPlatform,
+    meta: RolloutMetadata,
+}
+
+impl CanaryMode {
```

Comment:
I'd suggest using `bon`'s `builder` macro for this, so you don't have to pass five args at once. Not blocking, can fix forward.